### PR TITLE
Merge second part of Ben's reviews

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: courses/intro-to-ada/intro_to_ada.rst [hunk 1]
```diff
@@ -128,29 +128,40 @@ can leverage in SPARK to help proving your programs correct.
...
+Ada is a multi-paradigm language with support for object orientation
+and some elements of functional programming, but its core is a simple, coherent
+procedural/imperative language akin to C or Pascal.
 
-    One important distinction with a language like C is that statements
-    and expressions are very clearly distinguished.
+.. admonition:: In other languages
+
+    One important distinction between Ada and a language like C is that
+    statements and expressions are very clearly distinguished.  In Ada, If you
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: "In Ada, If" => "if"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 2]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
-by default.
+This version utilizes an Ada feature known as a :ada:`use` clause, which has
+the the form ``use`` *package-name*. As illustrated by the call on
+:ada:`Put_Line`, the effect is that entities from the named package can be
+referenced directly, without the *package-name.* prefix.
+
+Imperative language - If/Then/Else
+----------------------------------
+
+This section describes Ada's ``if`` statement and introduces several other
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: ada role missing here

File: courses/intro-to-ada/intro_to_ada.rst [hunk 3]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+fundamental language facilities including integer I/O, data declarations,
+and subprogram parameter modes.
+
+Ada's ``if`` statementis pretty unsurprising in form and function:
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: ada role missing here, too.

File: courses/intro-to-ada/intro_to_ada.rst (hunk 2 above)
Comment (gusthoff) [reply courses/intro-to-ada/intro_to_ada.rst]: I'm seeing this role missing in multiple places. Is this distinction between "if" as a statement and as a keyword on purpose?

File: courses/intro-to-ada/intro_to_ada.rst [hunk 4]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+       Put ("Enter an integer value: ");  -- Put a String
+       Get (N);  -- Read in an integer value
+       if N>0 then
+          Put (N);  --Put an Integer
+          Put_Line (" is a positive number");
+       end if;
+    end Check_Positive;
+
+The ``if`` statement minimally consists of the reserved word :ada:`if`, a condition
+(which must be a boolean value), the reserved word :ada:`then` and a non-empty
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: "boolean" => "Boolean", as in "Gaussian", "Laplacian", "Ohmic", etc. 

Just for reference: we discussed this topic a few months ago.

File: courses/intro-to-ada/intro_to_ada.rst [hunk 5]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+sequence of statements (the ``then`` part) which is executed if the condition
+evaluates to True, and a terminating :ada:`end if`.
+
+This example declares an Integer variable N, prompts the user for an Integer,
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: "Integer", on the other hand, should not be capitalized if it's not directly referring to the type. If it is, however, then the ada-role is missing.

File: courses/intro-to-ada/intro_to_ada.rst [hunk 6]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+checks if the value is positive and, if so, displays the integer's value
+followed by the string " is a positive number". If the value is not positive,
+the procedure does not display any output.
+
+The type Integer is a predefined signed type, and its range depends on the
+computer architecture. On typical current processors Integer is 32-bit signed.
+
+The example illustrates some of the basic functionality for integer input-output.
+The relevant subprograms are in the predefined package Ada.Integer_Text_IO and
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue for package.

File: courses/intro-to-ada/intro_to_ada.rst [hunk 7]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
--  ``1 .. 10`` is a discrete range, from ``1`` to ``10`` included.
+-  ``1 .. 5`` is a discrete range, from ``1`` to ``5`` inclusive.
+
+-  The loop parameter ``I`` (the name is arbitrary) in the body of the
+   loop has a value within this range.
+
+-  ``I`` is local to the loop, so you cannot refer to ``I``
+   outside the loop.
+
+-  Although the value of `I`` is incremented at each iteration, from the
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: missing back quote mark for "I"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 8]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+   program's perspective it is constant. An attempt to modify its value
+   is illegal; the compiler would reject the program.
 
--  It is bound to the name ``I`` in the body of the loop.
+-  Integer'Image is a function that takes an Integer and converts it to a
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Missing ada-role

File: courses/intro-to-ada/intro_to_ada.rst [hunk 9]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+   String.  It is an example of a language construct known as an *attribute*,
+   indicated by the "'" syntax, which will be covered in more detail later.
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: replaced double-quotes by back-quotes.

File: courses/intro-to-ada/intro_to_ada.rst [hunk 10]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+       while I <= 5 loop
+          Put_Line ("Hello, World!" & Integer'Image(I));
 
-          --  Assignment
           I := I + 1;
        end loop;
-    end Greet;
+    end Greet_5c;
+
+The condition is evaluated before each iteration. If it is False,
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: missing ada-role for "False"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 11]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+then the loop is terminated.
+
+This program has the same effect as the previous examples.
+
+.. admonition:: In other languages
 
-Here we see what assignment to a variable looks like. There is no
-``I++`` short form to increment, as there is in many languages.
+    Note that Ada has different semantics than C-based languages with respect
+    to the condition in a while loop.  In Ada the condition has to be a boolean
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: "boolean" => "Boolean"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 12]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+    value or the compiler will reject the program; the condition is not an
+    integer that is treated as either True or False depending on whether it is
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: missing ada-role (or back-quotes) for "True" / "False"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 13]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+             when others =>
+                Put_Line (" Au revoir");
+                exit;
+          end case;
+       end loop;
+    end Check_Direction;
 
-             when 3 .. 5 =>
-                Put_Line ("Hello");
+This program repeatedly prompts for an Integer value and then, if the value is
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: back-quotes for "Integer"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 14]
```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
...
+   either an integer type or an enumeration type.  Discrete types will
+   be covered in more detail later
+   `discrete types <TODO:linktodiscretetypes>`__.
 
-Notable points about Ada's case statement:
+-  Every possible value for the case expression needs to be covered by a unique
+   branch of the case statement. This will be checked at compile time.
+
+-  A branch can specify a single value, such as ``0``; a range of values,
+   such as ``1..89``; or any combination of the two (separated by a "|"
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Shouldn't we use whitespaces for ranges?

File: courses/intro-to-ada/intro_to_ada.rst [hunk 15]
```diff
@@ -460,9 +643,10 @@ Ada. What is important to know at this stage:
        --  Call to Nested
     end Main;
 
--  You cannot declare anything outside of a declarative region. If you
-   need to scope variables in a subprogram, you can introduce a new
-   declarative region with the :ada:`declare` block
+-  A declaration cannot appear as a statement If you
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: "If" => "if"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 16]
```diff
@@ -460,9 +643,10 @@ Ada. What is important to know at this stage:
...
+   need to declare a local variable amidst the statements,
+   you can introduce a new declarative region with a :ada:`block`
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: wrong ada-role here...

File: courses/intro-to-ada/intro_to_ada.rst [hunk 17]
```diff
@@ -488,48 +672,66 @@ Ada. What is important to know at this stage:
...
+    The Get_Line function allows you to receive input from the user, and get
+    the result as a string. It is more or less equivalent to the :c:`scanf`
+    C function.
 
     It returns a String, which, as we will see later, is an
-    :ref:`Unconstrained array type <UnconstrainedArrayTypes>`. For the moment,
-    it is sufficient to understand that you must declare the ``Name`` string
-    variable and initialize it at the same time.
+    :ref:`Unconstrained array type <UnconstrainedArrayTypes>`.
+    For now we simply note that, if you wish to declare a String
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: missing ada-role for "String"

File: courses/intro-to-ada/intro_to_ada.rst [hunk 18]
```diff
@@ -488,48 +672,66 @@ Ada. What is important to know at this stage:
...
-                       elsif A = 13 then 15
-                       else 18);
+    procedure Check_Positive is
+       N : Integer;
     begin
-       null;  --  When a subprogram is empty, null statement is mandatory
-    end Main;
+       Put ("Enter an integer value: ");  -- Put a String
+       Get (N);  -- Reads in an integer value
+       Put (N);  --Put an Integer
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: missing whitespace in comment.

File: courses/intro-to-ada/intro_to_ada.rst [hunk 19]
```diff
@@ -488,48 +672,66 @@ Ada. What is important to know at this stage:
...
+                             " is not a positive number" );
+       begin
+          Put_Line (S);
+       end;
+    end Check_Positive;
 
-Ada's if expression are similar to if statements. However, there are a few
-differences that stems from the fact that it is an expression:
+The :ada:`if` expression evaluates to one of the two Strings depending
+on N, and assigns that value to the local variable S.
```
Comment (gusthoff) [inline courses/intro-to-ada/intro_to_ada.rst]: Formatting issue: missing back-quotes for variables.

File: courses/intro-to-ada/intro_to_ada.rst (hunk 14 above)
Comment (raph-amiard) [reply courses/intro-to-ada/intro_to_ada.rst]: Yes!

**Comment by raph-amiard** [comment]
Hello Gustavo, I agree with everything you said! However, could you take care of submitting a new version of this PR with your edits ? It's a lot easier when there are lot of small changes like that if you just do them directly, especially non controversial low level things like that :)

TIA!

**Pull request merged**